### PR TITLE
[HttpKernel] Add `#[MapQueryParameter]`, `#[MapQueryString]` and `#[MapRequestPayload]`

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Back at it to fulfill the `Attributes Overview` document :smile: 

Resolves https://github.com/symfony/symfony-docs/issues/18216
Resolves https://github.com/symfony/symfony-docs/issues/18206

### CODE REVIEW

File: controller.rst [hunk 1]
```diff
@@ -336,6 +336,157 @@ object. To access it in your controller, add it as an argument and
...
+.. code-block:: json
+
+    {
+        "firstName": "John",
+        "lastName": "Smith",
+        "age": 28
+    }
+
+In this case, it is also possible to directly map this payload to your DTO thanks
+to the :class:`Symfony\\Component\\HttpKernel\\Attribute\\MapRequestPayload`
```
Comment (OskarStark) [inline controller.rst]:
```suggestion
In this case, it is also possible to directly map this payload to your DTO by
using the :class:`Symfony\\Component\\HttpKernel\\Attribute\\MapRequestPayload`
```

File: controller.rst [hunk 2]
```diff
@@ -336,6 +336,157 @@ object. To access it in your controller, add it as an argument and
...
+
+    public function dashboard(
+        #[MapRequestPayload] UserDTO $userDto
+    ): Response
+    {
+        // ...
+    }
+
+This attribute allows you to customize the serialization context as well
+as the class responsible of doing the mapping between the request and
```
Comment (OskarStark) [inline controller.rst:471]:
```suggestion
as the class is responsible of doing the mapping between the request and
```

Comment (alexandre-daubois) [reply controller.rst:471]: I'm not sure about this :thinking: The sentence seems correct as is. Or maybe `as the class that is responsible`?

File: controller.rst [hunk 3]
```diff
@@ -336,6 +336,157 @@ object. To access it in your controller, add it as an argument and
...
+
+    The :class:`Symfony\\Component\\HttpKernel\\Attribute\\MapQueryParameter` attribute
+    was introduced in Symfony 6.3.
+
+Mapping The Whole Query String
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Another possibility is to map the entire query string into an object that will hold
+available query parameters. Let's say you declare the following DTO with its
+validation constraints::
```
Comment (94noni) [inline controller.rst]:
```suggestion
optional validation constraints::
```

to decouple the fact that we can map without any constraint rules :)

Comment (alexandre-daubois) [reply controller.rst]: Good suggestion in my opinion, added! :slightly_smiling_face:

File: controller.rst (hunk 2 above)
Comment (javiereguiluz) [reply controller.rst:471]: I agree with Alex on this.

**Comment by OskarStark** [comment]
Thanks for your work on this new feature!

**Pull request merged**